File: src/activities.js
// Anime image APIs — see discord-images.js for resolution pipeline
export const WAIFU_TAGS = {
  food: 'neko',
  gaming: 'shinobu',
  chill: 'megumin',
  work: 'awoo',
  social: 'cry',
};

// Re-export verified GIF fallbacks (nekos.best — Tenor URLs were 404)
export { VERIFIED_FALLBACKS as FALLBACK_GIFS, fetchWaifuImage, fetchNekosBest } from './discord-images.js';

export const ACTIVITY_CATEGORIES = [
  {
    id: 'food',
    label: 'Food',
    emoji: '🍽️',
    color: '#f7768e',
    waifuTag: 'neko',
    fallbackGif: 'https://nekos.best/api/v2/nom/eaa199e9-2b86-4b15-87d1-53688e36d8ec.gif',
    activities: [
      { id: 'eating-pizza', details: 'Eating', state: 'Pizza night 🍕', emoji: '🍕', nekosEndpoint: 'feed', largeImageKey: 'pizza', largeImageText: 'Pizza time' },
      { id: 'eating-sushi', details: 'Eating', state: 'Sushi run 🍣', emoji: '🍣', nekosEndpoint: 'nom', largeImageKey: 'sushi', largeImageText: 'Sushi' },
      { id: 'eating-ramen', details: 'Eating', state: 'Ramen bowl 🍜', emoji: '🍜', nekosEndpoint: 'feed', largeImageKey: 'ramen', largeImageText: 'Ramen' },
      { id: 'eating-burger', details: 'Eating', state: 'Burger time 🍔', emoji: '🍔', nekosEndpoint: 'bite', largeImageKey: 'burger', largeImageText: 'Burger' },
      { id: 'eating-tacos', details: 'Eating', state: 'Taco Tuesday 🌮', emoji: '🌮', nekosEndpoint: 'nom', largeImageKey: 'tacos', largeImageText: 'Tacos' },
      { id: 'eating-snacks', details: 'Snacking', state: 'Midnight munchies 🍿', emoji: '🍿', nekosEndpoint: 'nom', largeImageKey: 'snacks', largeImageText: 'Snacks' },
      { id: 'cooking', details: 'Cooking', state: 'Chef mode 👨‍🍳', emoji: '👨‍🍳', nekosEndpoint: 'feed', largeImageKey: 'cooking', largeImageText: 'Cooking' },
      { id: 'eating-dessert', details: 'Eating', state: 'Sweet tooth 🍰', emoji: '🍰', nekosEndpoint: 'bite', largeImageKey: 'dessert', largeImageText: 'Dessert' },
    ],
  },
  {
    id: 'gaming',
    label: 'Gaming',
    emoji: '🎮',
    color: '#7aa2f7',
    waifuTag: 'shinobu',
    fallbackGif: 'https://nekos.best/api/v2/yeet/ae4dda45-2175-4576-b957-58dcc1362284.gif',
    activities: [
      { id: 'gaming', details: 'Gaming', state: 'In the zone', emoji: '🎮', nekosEndpoint: 'yeet', largeImageKey: 'gaming', largeImageText: 'Gaming' },
      { id: 'ranked', details: 'Gaming', state: 'Ranked grind 🔥', emoji: '🔥', nekosEndpoint: 'yeet', largeImageKey: 'ranked', largeImageText: 'Ranked' },
      { id: 'coop', details: 'Gaming', state: 'Co-op with friends', emoji: '👥', nekosEndpoint: 'hug', largeImageKey: 'coop', largeImageText: 'Co-op' },
      { id: 'retro', details: 'Gaming', state: 'Retro classics 🕹️', emoji: '🕹️', nekosEndpoint: 'dance', largeImageKey: 'retro', largeImageText: 'Retro' },
      { id: 'speedrun', details: 'Gaming', state: 'Speedrunning ⏱️', emoji: '⏱️', nekosEndpoint: 'yeet', largeImageKey: 'speedrun', largeImageText: 'Speedrun' },
      { id: 'vr-gaming', details: 'Gaming', state: 'In VR 🥽', emoji: '🥽', nekosEndpoint: 'dance', largeImageKey: 'vr', largeImageText: 'VR Gaming' },
    ],
  },
  {
    id: 'chill',
    label: 'Chill',
    emoji: '😌',
    color: '#9ece6a',
    waifuTag: 'megumin',
    fallbackGif: 'https://nekos.best/api/v2/sleep/611a318f-1645-48f4-9cc0-099eb8d817d9.gif',
    activities: [
      { id: 'sleeping', details: 'Sleeping', state: 'Do not disturb 💤', emoji: '💤', nekosEndpoint: 'sleep', largeImageKey: 'sleep', largeImageText: 'Sleeping' },
      { id: 'napping', details: 'Napping', state: 'Power nap mode', emoji: '😴', nekosEndpoint: 'sleep', largeImageKey: 'nap', largeImageText: 'Napping' },
      { id: 'reading', details: 'Reading', state: 'Lost in a book 📚', emoji: '📚', nekosEndpoint: 'smile', largeImageKey: 'reading', largeImageText: 'Reading' },
      { id: 'listening', details: 'Listening to music', state: 'Vibing 🎧', emoji: '🎧', nekosEndpoint: 'dance', largeImageKey: 'music', largeImageText: 'Music' },
      { id: 'meditating', details: 'Meditating', state: 'Finding peace 🧘', emoji: '🧘', nekosEndpoint: 'smile', largeImageKey: 'meditate', largeImageText: 'Meditating' },
      { id: 'bath', details: 'Relaxing', state: 'Bubble bath 🛁', emoji: '🛁', nekosEndpoint: 'happy', largeImageKey: 'bath', largeImageText: 'Relaxing' },
    ],
  },
  {
    id: 'work',
    label: 'Work',
    emoji: '💻',
    color: '#bb9af7',
    waifuTag: 'awoo',
    fallbackGif: 'https://nekos.best/api/v2/bored/82f8fec0-d651-4905-a739-5917d728f89f.gif',
    activities: [
      { id: 'coding', details: 'Coding', state: 'Building something cool', emoji: '💻', preferWaifu: true, waifuTag: 'awoo', largeImageKey: 'coding', largeImageText: 'Coding' },
      { id: 'studying', details: 'Studying', state: 'Brain gains 📖', emoji: '📖', nekosEndpoint: 'bored', largeImageKey: 'studying', largeImageText: 'Studying' },
      { id: 'meeting', details: 'In a meeting', state: 'Busy until further notice', emoji: '📅', nekosEndpoint: 'wave', largeImageKey: 'meeting', largeImageText: 'Meeting' },
      { id: 'focus', details: 'Deep focus', state: 'Heads down 🔒', emoji: '🔒', nekosEndpoint: 'bored', largeImageKey: 'focus', largeImageText: 'Focus mode' },
      { id: 'designing', details: 'Designing', state: 'Creating art 🎨', emoji: '🎨', nekosEndpoint: 'smile', largeImageKey: 'design', largeImageText: 'Designing' },
      { id: 'writing', details: 'Writing', state: 'Words flowing ✍️', emoji: '✍️', nekosEndpoint: 'bored', largeImageKey: 'writing', largeImageText: 'Writing' },
    ],
  },
  {
    id: 'social',
    label: 'Social',
    emoji: '✨',
    color: '#ff9e64',
    waifuTag: 'cry',
    fallbackGif: 'https://nekos.best/api/v2/wave/e6f276a8-11f1-4ad0-b1e0-3fa91678e2f4.gif',
    activities: [
      { id: 'streaming', details: 'Streaming', state: 'Live now 📺', emoji: '📺', nekosEndpoint: 'wave', largeImageKey: 'streaming', largeImageText: 'Streaming' },
      { id: 'watching', details: 'Watching', state: 'Movie night 🎬', emoji: '🎬', nekosEndpoint: 'happy', largeImageKey: 'watching', largeImageText: 'Watching' },
      { id: 'traveling', details: 'Traveling', state: 'On an adventure ✈️', emoji: '✈️', nekosEndpoint: 'wave', largeImageKey: 'travel', largeImageText: 'Traveling' },
      { id: 'gym', details: 'At the gym', state: 'Gains incoming 💪', emoji: '💪', nekosEndpoint: 'yeet', largeImageKey: 'gym', largeImageText: 'Gym' },
      { id: 'partying', details: 'Partying', state: 'Living my best life 🎉', emoji: '🎉', nekosEndpoint: 'dance', largeImageKey: 'party', largeImageText: 'Partying' },
      { id: 'shopping', details: 'Shopping', state: 'Retail therapy 🛍️', emoji: '🛍️', nekosEndpoint: 'happy', largeImageKey: 'shopping', largeImageText: 'Shopping' },
    ],
  },
];

export const ALL_ACTIVITIES = ACTIVITY_CATEGORIES.flatMap((c) =>
  c.activities.map((a) => ({
    ...a,
    category: c.id,
    categoryColor: c.color,
    waifuTag: a.waifuTag || c.waifuTag,
    fallbackGif: a.fallbackGif || c.fallbackGif,
  }))
);

File: legacy/smiley-v8/src-tauri/src/activities.rs
use crate::models::{Activity, Category};

fn a(
    id: &str,
    details: &str,
    state: &str,
    emoji: &str,
    category: &str,
    color: &str,
    gif: &str,
) -> Activity {
    Activity {
        id: id.into(),
        details: details.into(),
        state: state.into(),
        emoji: emoji.into(),
        category: category.into(),
        color: color.into(),
        gif: gif.into(),
    }
}

pub fn categories() -> Vec<Category> {
    let food = "#f7768e";
    let gaming = "#7aa2f7";
    let chill = "#9ece6a";
    let work = "#bb9af7";
    let social = "#ff9e64";

    let gaming_gif = "https://media.tenor.com/yjGe52tfF-wAAAAM/gaming-gamer.gif";
    let chill_gif = "https://media.tenor.com/BsoscZUHi-gAAAAM/sleepy-sleep.gif";
    let work_gif = "https://media.tenor.com/QLh0PhunTj8AAAAM/anime-typing.gif";
    let social_gif = "https://media.tenor.com/_EzjRj8XOP4AAAAM/streaming-stream.gif";

    vec![
        Category {
            id: "food".into(),
            label: "Food".into(),
            emoji: "🍽️".into(),
            color: food.into(),
            activities: vec![
                a("eating-pizza", "Eating", "Pizza night 🍕", "🍕", "food", food, "https://media.tenor.com/i-xS-A_DTCEAAAAM/pizza-food.gif"),
                a("eating-sushi", "Eating", "Sushi run 🍣", "🍣", "food", food, "https://media.tenor.com/KE361QFenNcAAAAM/anime-refei%C3%A7%C3%A3o-jap%C3%A3o-comida.gif"),
                a("eating-ramen", "Eating", "Ramen bowl 🍜", "🍜", "food", food, "https://media.tenor.com/3hCp28Y4JcUAAAAM/hungry-ramen.gif"),
                a("eating-burger", "Eating", "Burger time 🍔", "🍔", "food", food, "https://media.tenor.com/uk9xO0xpWoIAAAAM/burger-eating.gif"),
                a("cooking", "Cooking", "Chef mode 👨‍🍳", "👨‍🍳", "food", food, "https://media.tenor.com/flX5arjPeDcAAAAM/sora-cooking.gif"),
                a("eating-dessert", "Eating", "Sweet tooth 🍰", "🍰", "food", food, "https://media.tenor.com/DTRz6D1e5ZEAAAAM/eating-dessert-happily.gif"),
                a("eating-tacos", "Eating", "Taco Tuesday 🌮", "🌮", "food", food, "https://media.tenor.com/tz1kb3yen6wAAAAM/uwu-taco.gif"),
                a("eating-snacks", "Snacking", "Midnight munchies 🍿", "🍿", "food", food, "https://media.tenor.com/gBrP7QayoRkAAAAM/himouto-umaru-chan.gif"),
            ],
        },
        Category {
            id: "gaming".into(),
            label: "Gaming".into(),
            emoji: "🎮".into(),
            color: gaming.into(),
            activities: vec![
                a("gaming", "Gaming", "In the zone", "🎮", "gaming", gaming, gaming_gif),
                a("ranked", "Gaming", "Ranked grind 🔥", "🔥", "gaming", gaming, "https://media.tenor.com/o52AZQZ_PloAAAAM/kick-anime.gif"),
                a("coop", "Gaming", "Co-op with friends", "👥", "gaming", gaming, "https://media.tenor.com/ZIlcnod9hnkAAAAM/anime-anime-hug.gif"),
                a("retro", "Gaming", "Retro classics 🕹️", "🕹️", "gaming", gaming, "https://media.tenor.com/L99ayPjM6m4AAAAM/zyclunt-blue.gif"),
                a("speedrun", "Gaming", "Speedrunning ⏱️", "⏱️", "gaming", gaming, "https://media.tenor.com/V2FWyvBLJ0kAAAAM/anime-run-run.gif"),
                a("vr-gaming", "Gaming", "In VR 🥽", "🥽", "gaming", gaming, "https://media.tenor.com/qIvEeou-1FIAAAAM/play-network-anime-girl.gif"),
            ],
        },
        Category {
            id: "chill".into(),
            label: "Chill".into(),
            emoji: "😌".into(),
            color: chill.into(),
            activities: vec![
                a("sleeping", "Sleeping", "Do not disturb 💤", "💤", "chill", chill, chill_gif),
                a("napping", "Napping", "Power nap mode", "😴", "chill", chill, "https://media.tenor.com/jfwf7xpv5p0AAAAM/sleep-anime.gif"),
                a("reading", "Reading", "Lost in a book 📚", "📚", "chill", chill, "https://media.tenor.com/rJxGy9CYwHoAAAAM/anime-read.gif"),
                a("listening", "Listening to music", "Vibes on 🎵", "🎧", "chill", chill, "https://media.tenor.com/dN976uhxB0kAAAAM/aimoto-rinku-listening-to-music.gif"),
                a("meditating", "Meditating", "Finding peace 🧘", "🧘", "chill", chill, "https://media.tenor.com/7tcxu-UKvO8AAAAM/meditate-zen.gif"),
                a("bath", "Relaxing", "Bubble bath 🛁", "🛁", "chill", chill, "https://media.tenor.com/M3nkdB81tkQAAAAM/virgin-road-anime-relaxed.gif"),
            ],
        },
        Category {
            id: "work".into(),
            label: "Work".into(),
            emoji: "💻".into(),
            color: work.into(),
            activities: vec![
                a("coding", "Coding", "Building something cool", "💻", "work", work, work_gif),
                a("studying", "Studying", "Brain gains 📖", "📖", "work", work, "https://media.tenor.com/etfl8OlhPIYAAAAM/studying-anime-girl.gif"),
                a("meeting", "In a meeting", "Busy until further notice", "📅", "work", work, "https://media.tenor.com/_9W9bVa4AHgAAAAM/wavi-anime.gif"),
                a("focus", "Deep focus", "Heads down 🔒", "🔒", "work", work, work_gif),
                a("designing", "Designing", "Creating art 🎨", "🎨", "work", work, "https://media.tenor.com/GB_tiifwEJ0AAAAM/drawing-kaoruko-moeta.gif"),
                a("writing", "Writing", "Words flowing ✍️", "✍️", "work", work, "https://media.tenor.com/OeaK9o_vzMIAAAAM/anime-write.gif"),
            ],
        },
        Category {
            id: "social".into(),
            label: "Social".into(),
            emoji: "✨".into(),
            color: social.into(),
            activities: vec![
                a("streaming", "Streaming", "Live now 📺", "📺", "social", social, social_gif),
                a("watching", "Watching", "Movie night 🎬", "🎬", "social", social, "https://media.tenor.com/7_IxCInZUPQAAAAM/menhera-eating-popcorn.gif"),
                a("traveling", "Traveling", "On an adventure ✈️", "✈️", "social", social, "https://media.tenor.com/gPjII19ICdIAAAAM/road-road-trip-move-dragon-ball-anime-tyan-vibe-car.gif"),
                a("gym", "At the gym", "Gains incoming 💪", "💪", "social", social, "https://media.tenor.com/uKWmvSxGj3gAAAAM/gym-wow.gif"),
                a("partying", "Partying", "Living my best life 🎉", "🎉", "social", social, "https://media.tenor.com/ymPYRZ4YGbEAAAAM/partyhard-party.gif"),
                a("shopping", "Shopping", "Retail therapy 🛍️", "🛍️", "social", social, "https://media.tenor.com/9M34adQOtNwAAAAM/shopping-hi.gif"),
            ],
        },
    ]
}

pub fn find(id: &str) -> Option<Activity> {
    for c in categories() {
        if let Some(act) = c.activities.into_iter().find(|x| x.id == id) {
            return Some(act);
        }
    }
    None
}
